File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/VpsClientLink.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * A navigation link for VPS-related actions in the client portal.
 */
class VpsClientLink
{
    /**
     * Display label for the link.
     */
    #[DTA\Data(field: "label", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $label = null;

    /**
     * URL or route for the action.
     */
    #[DTA\Data(field: "link", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $link = null;

    /**
     * Icon class for the link.
     */
    #[DTA\Data(field: "icon", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $icon = null;

    /**
     * Icon text label.
     */
    #[DTA\Data(field: "icon_text", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $icon_text = null;

    /**
     * Help tooltip text for the link.
     */
    #[DTA\Data(field: "help_text", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $help_text = null;

}

File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/VpsClientLink.scala
package io.swagger.server.model


/**
 * A navigation link for VPS-related actions in the client portal.
 *
 * @param label Display label for the link.
 * @param link URL or route for the action.
 * @param icon Icon class for the link.
 * @param icon_text Icon text label.
 * @param help_text Help tooltip text for the link.
 */
case class VpsClientLink (
  label: Option[String],
  link: Option[String],
  icon: Option[String],
  icon_text: Option[String],
  help_text: Option[String]
)

